common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -1914,6 +1914,44 @@
     ],
     "sqlState" : "22012"
   },
+  "INTERVAL_ERROR" : {
+    "message" : [
+      "Interval error."
+    ],
+    "subClass" : {
+      "DAY_TIME_PARSING" : {
+        "message" : [
+          "Error parsing interval day-time string: <msg>."
+        ]
+      },
+      "ILLEGAL_DAY_OF_WEEK" : {
+        "message" : [
+          "Illegal input for day of week: <string>."
+        ]
+      },
+      "INTERVAL_PARSING" : {
+        "message" : [
+          "Error parsing interval <interval> string: <msg>."
+        ]
+      },
+      "SECOND_NANO_FORMAT" : {
+        "message" : [
+          "Interval string does not match second-nano format of ss.nnnnnnnnn."
+        ]
+      },
+      "UNMATCHED_FORMAT_STRING" : {
+        "message" : [
+          "Interval string does not match <intervalStr> format of <supportedFormat> when cast to <typeName>: <input><fallBackNotice>."
```

Comment:
We should avoid embedding error text to source code because:
1. Tech writers should be able to edit error text w/o modifying source codes.
2. In the future we could support errors in different languages. So, hard coding errors in English is not good idea.

Let's introduce separate error sub-conditions per every notice.

Reply:
Please, embed the text:
```scala
  private val fallbackNotice = s"set ${SQLConf.LEGACY_FROM_DAYTIME_STRING.key} to true " +
    "to restore the behavior before Spark 3.0."
```